examples/simulation_examples/rotation.py

```diff
@@ -0,0 +1,105 @@
+)
+from flow360.component.simulation.outputs import SurfaceOutput
+from flow360.component.simulation.physics_components import (
+    NavierStokesSolver,
+    SpalartAllmaras,
+)
+from flow360.component.simulation.primitives import Cylinder
+from flow360.component.simulation.references import ReferenceGeometry
+from flow360.component.simulation.simulation import SimulationParams
+from flow360.component.simulation.surfaces import Surface, Wall
+from flow360.component.simulation.time_stepping import SteadyTimeStepping
+from flow360.component.simulation.volumes import Rotation
+
+wing_surface = Surface(mesh_patch_name="1")
+
+rotation_zone_inner = Cylinder(
+    axis=(1, 1, 0), center=(0, 0, 0), height=1, inner_radius=0, outer_radius=2
+)
+
+rotation_zone_outer = Cylinder(
+    axis=(0, 1, 0), center=(0, 0, 0), height=4, inner_radius=0, outer_radius=5
+)
+
+
+with SI_unit_system:
+    simulationParams = SimulationParams(
+        # Global settings, skiped in current example
+        meshing=MeshingParameters(
+            refinement_factor=1,
+            farfield=Farfield(type="auto"),
+            refinements=[
+                CylindricalRefinement(
+                    entities=[rotation_zone_inner],
+                    # enclosed_objects=[wing_surface], # this will be automatically populated by analysing topology
+                    spacing_axial=0.1,
+                    spacing_radial=0.1,
+                    spacing_circumferential=0.2,
+                ),
+                CylindricalRefinement(
+                    entities=[rotation_zone_outer],
+                    # enclosed_objects=[rotation_zone_inner], # this will be automatically populated by analysing topology
+                    spacing_axial=0.1,
+                    spacing_radial=0.1,
+                    spacing_circumferential=0.2,
+                ),
+            ],
+        ),
+        operating_condition=ExternalFlowOperatingConditions(  # Per volume override
+            Mach=0.84, temperature=288.15, alpha=1.06 * u.deg, beta=1e-2 * u.rad
+        ),
+        reference_geometry=ReferenceGeometry(  # Per volume override
+            area=2,
+            moment_length=(1, 1, 1),
+            mesh_unit=1 * u.m,
+        ),
+        volumes=[
+            FluidDynamics(
+                entities=["*"],  # This means we apply settings to all the volume zones
+                navier_stokes_solver=NavierStokesSolver(),
+                turbulence_model_solver=SpalartAllmaras(),
+                operating_condition=ExternalFlowOperatingConditions(
+                    Mach=0.3,
+                    temperature=288.15,
+                ),
+                reference_geometry=ReferenceGeometry(
+                    area=1,
+                    moment_length=2,
+                    mesh_unit=3 * u.m,
+                ),
+            ),
+            Rotation(
+                entities=[rotation_zone_inner],
+                angular_velocity=5 * u.rpm,
+            ),
```

Comment:
Rotation shouldn't be a type of volume right? It's more like a property of a volume. Currently in json it's defined under FluidDynamics type of volume zones. Why didn't we go with that design?

Reply:
We renamed `volume` to `models` which suits the Rotation concept better. `volumes` created a lot of confusion.

"FluidDynamics" is not a good name for rotating zone. BETDisk is also "FluidDynamics" same for Actuator disk and so on.

Also I think nesting `rotation` inside `FludiDynamics` is not good design, it creates unnecessary burden to write FludiDynamics every time when the Rotation itself already implies that.

Reply:
I see. This makes sense. What's the consideration of putting navier_stokes_solver and turbulent_model_solver definitions under FluidDynamics? It's possible that the user does not define a FluidDynamics model but uses only Rotation in the case of SRF, or only PorousMedia in the case of a single porous zone. Should these classes also have solver definitions as their attributes?
It doesn't make sense to me to enable the possibility of individual navier stokes solver configurations for each FluidDynamics or Rotation model though. These zones are strongly coupled. I don't think It's likely that in the future we will need such functionality either

Reply:
I think by default we will have one `FluidDynamics` applied to all volumes if the user did not specify one. Or we throw error.

We had a previous design with NS and turb solver in all models and got comment from application team that each model should focus on attributes that are specific to that model.

I agree with you that we do not see the need for changing NS config in each zone at this moment but the main goal of this design is to have an interface that is _as future-proof as possible_. It will be very painful to change interface and consider compatibility issues when we realize some time in the future that some new parameter in NS/SA/SST etc DO need to change with zones.

Reply:
I think even if we need to have solver configuration change with zones in the future, the interface should be solver config owning zones, not the other way around. Let's discuss tomorrow